x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/event_log_adapter/event_log_adapter.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IEventLogService } from '../../../../../../event_log/server';
+import {
+  FindBulkExecutionLogArgs,
+  FindExecutionLogArgs,
+  IRuleExecutionLogClient,
+  LogExecutionMetricsArgs,
+  LogStatusChangeArgs,
+  UpdateExecutionLogArgs,
+} from '../types';
+import { EventLogClient } from './event_log_client';
+
+export class EventLogAdapter implements IRuleExecutionLogClient {
+  private eventLogClient: EventLogClient;
+
+  constructor(eventLogService: IEventLogService) {
+    this.eventLogClient = new EventLogClient(eventLogService);
+  }
+
+  public async find({ ruleId, logsCount = 1, spaceId }: FindExecutionLogArgs) {
+    return []; // TODO Implement
+  }
+
+  public async findBulk({ ruleIds, logsCount = 1, spaceId }: FindBulkExecutionLogArgs) {
+    return {}; // TODO Implement
+  }
+
+  public async update({ attributes, spaceId }: UpdateExecutionLogArgs) {
+    // execution events are immutable, so we just log a status change istead of updating previous
+    if (attributes.status) {
+      await this.eventLogClient.logStatusChange({
+        ruleId: attributes.alertId,
+        newStatus: attributes.status,
+        spaceId,
+      });
+    }
+  }
+
+  public async delete(id: string) {
+    // execution events are immutable, nothing to do here
+  }
+
+  public async logExecutionMetrics({ ruleId, spaceId, metrics }: LogExecutionMetricsArgs) {
+    return this.eventLogClient.logExecutionMetrics({
+      ruleId,
+      spaceId,
+      metrics: {
+        executionGapDuration: metrics.executionGap?.asSeconds(),
+        indexingDurationSum: metrics.indexingDurations?.reduce((acc, cur) => acc + Number(cur), 0),
+        searchDurationSum: metrics.searchDurations?.reduce((acc, cur) => acc + Number(cur), 0),
+      },
+    });
+  }
+
+  public async logStatusChange(args: LogStatusChangeArgs) {
+    if (args.metrics) {
+      this.logExecutionMetrics({
+        ruleId: args.ruleId,
+        spaceId: args.spaceId,
+        metrics: args.metrics,
+      });
+    }
```

Comment:
Await?

Reply:
I'll need to poke, but in general, the functions which are used to add an event document - and this looks like one - are sync, not async.  You don't need to `await`.  We maintain a buffer of documents which we flush on a size / time basis.  So the log() type of functions are really just adding the doc to the buffer ...

Reply:
Yup, `await` is not needed here; all write operations are sync.